cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -75,6 +77,47 @@ void print_node(const std::string &prefix,
   }
 }
 
+template <typename T>
+std::string to_string_high_precision(T x) {
+  static_assert(std::is_floating_point<T>::value || std::is_integral<T>::value,
+                "T must be float, double, or integer");
+  std::ostringstream oss;
+  oss.imbue(std::locale::classic());  // use C locale
```

Comment:
Would it help to import a full C++ json library? I know DMLC was hesitant on this, but it could make this API simpler and help with others in the future. I was thinking like RapidJson (which is header-only): https://github.com/Tencent/rapidjson/

Reply:
We have to consider the trade-off for introducing a new dependency. I personally feel that the full-fledged JSON library is an overkill for what we're doing here, which is to generate a JSON dump.

We'd have a more compelling argument for a JSON library, if we are using JSON as a **serialization format**, i.e. storage and exchange format for cuML RF models. In that use case, round-trip performance and robustness becomes a concern.

Reply:
Since cuML uses Conda package manager, it might be easier to introduce 3rd-party dependencies.

Reply:
Decided to hold off on this now per discussion